pkg/dns/serviceresolver.go

```diff
@@ -115,13 +119,14 @@ func (b *ServiceResolver) Records(dnsName string, exact bool) ([]msg.Service, er
 				if fallback, ok := b.fallback(prefix, exact); ok {
 					return b.Records(fallback+b.base, exact)
 				}
+				return nil, errNoSuchName
 			}
-			return nil, err
+			return nil, errNoSuchName
 		}
 
 		// no portalIP and not headless, no DNS
 		if len(svc.Spec.ClusterIP) == 0 {
-			return nil, nil
+			return nil, errNoSuchName
```

Comment:
in cases where we recognize the service, but don't want to return records, will returning `errNoSuchName` trigger a recursive lookup we don't want? seems like we'd actually want to stop in this case and return no records